src/Init/Data/Nat/Basic.lean

```diff
@@ -571,12 +620,18 @@ attribute [simp] zero_lt_succ
 
 theorem succ_ne_self (n) : succ n ≠ n := Nat.ne_of_gt (lt_succ_self n)
 
+theorem add_one_ne_self (n) : n + 1 ≠ n := Nat.ne_of_gt (lt_succ_self n)
+
 theorem succ_le : succ n ≤ m ↔ n < m := .rfl
 
+theorem add_one_le_iff' : n + 1 ≤ m ↔ n < m := .rfl
```

Comment:
Similarly here.

Reply:
This also exists in Mathlib with the same name.